babble/nlp/engine.py

```diff
@@ -134,10 +135,11 @@ def evaluate(self, phrase: str) -> Optional[Understanding]:
         """Returns the Understanding of the given phrase. If phrase could not
         be understood None is returnd"""
 
-        # Try to match the given phrase with all intents. As soon as a matching
-        # intent is found return it.
+        # Try to match the given phrase with intents. Intents are filtered based on rule length.
+        #  As soon as a matching intent is found return it.
```

Comment:
```suggestion
        # Try to match the given phrase with intents.
        #
        # For performance improvements intents are filtered based on rule length 
        # so that rules of intent matches nearly the length of the phrase. 
        # Rules which are too long or short might match, but are not taken into 
        # account anyway because of the validity calculation of the match.
        #
        # As soon as a matching intent is found return it.
```